### PR TITLE
Feature - Design of /secrets raml and client

### PR DESCRIPTION
**Pull request by rqc**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/eedevops/he-design/pull/14" target="_blank">ReviewNinja

File: auth/auth-service.raml [hunk 1]
```diff
@@ -43,7 +45,7 @@ version: v1
         in the types section.
 
       responses:
-        204:
+        200:
```
Comment (gmatoshp) [inline auth/auth-service.raml]: maybe have also a response 500 available for all endpoints?

File: auth/client.md [hunk 2]
```diff
@@ -100,69 +100,43 @@ authClient = authService.Client(endpoint, params)
...
 
+  * **Examples**
+    > TODO
 
 ### `AuthServiceClient.authenticated(user, integration, opts, callback)`
 
+  * **Description**
+
+    > Wrapper for [`getSecrets()`](#authserviceclient.getsecrets).
+
```
Comment (gmatoshp) [inline auth/client.md]: need to define what the opts parameter is.

File: auth/secrets.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  // execute basic auth
+
+robot.e.registerIntegration(
+  { auth: { callback: myAuthCallback ...} }
+)
+
+// in hubot-enterprise auth flow
+secrets = AuthServiceClient.getSecrets(user, integration, integration.callback)
+
+```
```
Comment (gmatoshp) [inline auth/secrets.md]: looks good for now, once the "TODOs" are filled it will look even better.

File: auth/client.md (hunk 2 above)
Comment (rqc) [inline auth/client.md]: - [ ] it should have no opts parameter. Thanks!

File: auth/secrets.md (hunk 3 above)
Comment (rqc) [inline auth/secrets.md]: @gmatoshp Thanks, the `TODOs` will be a addressed incrementally in following PRs. Thanks for the review!

**Comment by rqc** [comment]
I am merging this. Please submit any problems as issues. Thanks 😄 !

**Pull request merged**